### PR TITLE
DLPJTS-116 custom signature label

### PR DESCRIPTION
**Pull request by datalogics-kam**
#### Changes in this pull request

- Use a unicode character in the signature.
- Embed a font and show that Type 0 fonts are subset.

#### Checklist for approving this pull request

(PR creator amend this with more conditions if necessary)

- [x] There are **unit tests** for new/changed code, or a good explanation why this was not possible.
- [x] All **CI builders** have indicated success (Give them a few minutes to notice the pull request.)
- [x] The **Pull request title** has the JIRA issue numbers separated by spaces (if any), a space, and then a short, but descriptive summary.
- [x] **Commit messages** are well formed: [A note about Git commit messages](http://www.tpope.net/node/106)
- [x] New public packages, classes, and methods are **documented**. (Strongly consider documenting private classes and methods.)

#### Blockers

Add a checkbox for yourself with your **&#64;name** to this list if you are holding this PR open for review. PR Shepherd, please hold off merging this PR until these are all checked:

- [x] _&lt;add your name here with an **@**>_

### CODE REVIEW

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-devauto** [comment]
Build failed.

**Comment by datalogics-kam** [comment]
trigger build

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-kam** [comment]
@datalogics-coreyy This is ready to go.

File: src/main/java/com/datalogics/pdf/samples/signature/SignDocument.java [hunk 1]
```diff
@@ -74,6 +74,17 @@
     private static final Double MM_PER_INCH = 25.4; // millimeters per inch
     private static final Double DEFAULT_MM_PER_PIXEL = 0.35277778; // default millimeters per pixel
 
+    // Customize the label. You can provide your own string that describes who
+    // signed the document.  The "{0}" will be replaced by the name of the
+    // signer, which is taken from the UserInfo.name in the
```
Comment (datalogics-bhaugen) [inline src/main/java/com/datalogics/pdf/samples/signature/SignDocument.java]: Looks like there is something missing from this sentence in the documentation

Comment (datalogics-kam) [reply src/main/java/com/datalogics/pdf/samples/signature/SignDocument.java]: Thanks. I fixed it up and also pulled up the signer name.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

File: src/main/java/com/datalogics/pdf/samples/util/DocumentUtils.java [hunk 2]
```diff
@@ -87,4 +105,75 @@ public static PDFDocument openPdfDocumentWithOptions(final URL inputUrl, final P
...
+     * @return a list of the content items
+     * @throws PDFInvalidDocumentException a general problem with the PDF document, which may now be in an invalid state
+     * @throws PDFIOException there was an error reading or writing a PDF file or temporary caches
+     * @throws PDFSecurityException some general security issue occurred during the processing of the request
+     * @throws IOException an I/O operation failed or was interrupted
+     * @throws PDFInvalidParameterException one or more of the parameters passed to a method is invalid
+     * @throws PDFFontException there was an error in the font set or an individual font
+     * @throws PDFConfigurationException there was a system problem configuring PDF support
+     */
+    public static List<RasterContentItem> getFormContentItems(final PDFPage page, final PDFXObjectForm form,
```
Comment (datalogics-coreyy) [inline src/main/java/com/datalogics/pdf/samples/util/DocumentUtils.java:146]: Nice! This is going to be really useful

File: src/test/java/com/datalogics/pdf/samples/util/ContentTextItemMatchers.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2015 Datalogics, Inc.
```
Comment (datalogics-coreyy) [inline src/test/java/com/datalogics/pdf/samples/util/ContentTextItemMatchers.java]: Date should be 2017

File: src/test/java/com/datalogics/pdf/samples/signature/SignDocumentTest.java [hunk 4]
```diff
@@ -53,11 +70,51 @@ public void testSignExistingSignatureFields() throws Exception {
...
+                                           hasText("this"),
+                                           hasText(" "),
+                                           hasText("document")));
+
         } finally {
             doc.close();
         }
     }
 
+    private List<ContentTextItem<?, ?>> getContentTextItems(XObject overlayTextForm) {
```
Comment (datalogics-coreyy) [inline src/test/java/com/datalogics/pdf/samples/signature/SignDocumentTest.java]: I might call 'overlayTextForm' something else here like 'signatureAppearanceForm' maybe?

File: src/main/java/com/datalogics/pdf/samples/signature/SignDocument.java (hunk 1 above)
Comment (datalogics-coreyy) [reply src/main/java/com/datalogics/pdf/samples/signature/SignDocument.java]: I think @datalogics-bhaugen was referring to line 79 ', which is taken from the UserInfo.name in the'

File: src/test/java/com/datalogics/pdf/samples/signature/SignDocumentTest.java [hunk 5]
```diff
@@ -43,64 +52,101 @@
...
+        pdfDocument = null;
+    }
 
-        final File file = SampleTest.newOutputFileWithDelete(FILE_NAME);
+    /**
+     * Create the document if this is the first test run in the class.
+     *
+     * @throws Exception a general exception was thrown
+     */
+    private static void ensureDocument() throws Exception {
```
Comment (datalogics-coreyy) [inline src/test/java/com/datalogics/pdf/samples/signature/SignDocumentTest.java]: I think you could call ensureDocument in setUpDocument. Then you can avoid calling ensureDocument 3 times for each test. They don't seem to be manipulating the document so it should be okay.

File: src/test/java/com/datalogics/pdf/samples/signature/SignDocumentTest.java (hunk 4 above)
Comment (datalogics-coreyy) [reply src/test/java/com/datalogics/pdf/samples/signature/SignDocumentTest.java]: Eh I guess it really doesn't matter...update at your discretion

**Review by datalogics-coreyy — CHANGES_REQUESTED** [review]

File: src/test/java/com/datalogics/pdf/samples/signature/SignDocumentTest.java (hunk 5 above)
Comment (datalogics-kam) [reply src/test/java/com/datalogics/pdf/samples/signature/SignDocumentTest.java]: Actually, I don't want to do all that work in `setUpDocument` for a few reasons:

* The tests are timed, but the setup is not, so you can't tell how much time it took to set up the document by looking in the IDE
* Errors thrown during setup are hard to understand and diagnose

So I just have the first test that runs make the document.

File: src/test/java/com/datalogics/pdf/samples/util/ContentTextItemMatchers.java (hunk 3 above)
Comment (datalogics-kam) [reply src/test/java/com/datalogics/pdf/samples/util/ContentTextItemMatchers.java]: Got it, thanks!

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Review by datalogics-coreyy — APPROVED** [review]

File: src/main/java/com/datalogics/pdf/samples/signature/SignDocument.java [hunk 6]
```diff
@@ -169,6 +183,10 @@ private static void signField(final SignatureManager sigMgr,
 
                     appearanceOptions.setFontSet(fontSet);
                     appearanceOptions.setDisplayItems(displayItems);
+
+                    // Customize the label. See the declaration of the constant for more info.
+                    appearanceOptions.setSignatureLabel(SIGNATURE_LABEL);
```
Comment (datalogics-coreyy) [inline src/main/java/com/datalogics/pdf/samples/signature/SignDocument.java:188]: Eclipse can't find this recently added 'setSignatureLabel'. I think you need to exclude this from the Talkeetna 5 build like we did with the export to csv sample.

**Review by datalogics-coreyy — CHANGES_REQUESTED** [review]

File: src/main/java/com/datalogics/pdf/samples/signature/SignDocument.java (hunk 6 above)
Comment (datalogics-kam) [reply src/main/java/com/datalogics/pdf/samples/signature/SignDocument.java:188]: Oooh, good catch.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Comment by datalogics-devauto** [comment]
Build succeeded.

**Review by datalogics-coreyy — APPROVED** [review]

**Pull request merged**